addons/stock/models/stock_move.py

```diff
@@ -1088,7 +1088,8 @@ def _onchange_move_line_ids(self):
             if breaking_char in (move_line.lot_name or ''):
                 split_lines = move_line.lot_name.split(breaking_char)
                 split_lines = list(filter(None, split_lines))
-                move_line.lot_name = split_lines[0]
```

Comment:
```diff
-                if split_lines:
-                    move_line.lot_name = split_lines[0]
+                move_line.lot_name = split_lines[0:]
```

Reply:
For me it looks good, but need a verification from someone who knows better stock than me

Reply:
LGTM too! You have to apply the changes proposed by Achraf and target V14 instead of V16.
Thanks for your fix!

Reply:
Sorry to interrupt this lgtm chain but actually this is wrong héhé
`[0:]` means "take all starting from the first one, inclusive"
Here we want to take the first one (if it exists), so "take all up to the second one, exclusive" -> `[:1]`

Reply:
@adwid nice catch, Thanks!

Reply:
Hello @adwid
What about `split_lines[:1]`, which means "take first char from string"? 

Reply:
Nop, it would return a sub-list, here we want a string (or False)
Actually your fix was already good, we could just consider my proposition as cleaner
Since you have to retarget the PR, I thought it was a good opportunity to apply my suggestion